sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -760,7 +762,14 @@ private[hive] case class InsertIntoHiveTable(
   override def children: Seq[LogicalPlan] = child :: Nil
   override def output: Seq[Attribute] = child.output
 
-  override lazy val resolved: Boolean = childrenResolved && child.output.zip(table.output).forall {
+  val numDynamicPartitions = partition.values.count(_.isEmpty)
+  val tableOutput = {
```

Comment:
If I understand correctly, this is the expected schema of the table that is generating the data to be inserted, possibly including dynamic partition columns.  Can you add a comment to that effect if I'm correct?